packages/react-components/src/shared/src/component-types.ts

```diff
@@ -0,0 +1,28 @@
+import { ElementType, ForwardedRef, PropsWithChildren, ReactNode } from "react";
+
+export interface ForwardedRefAttribute<T = any> {
+    /**
+     * @ignore
+     */
+    forwardedRef: ForwardedRef<T>
+}
+
+export interface AsAttribute<As extends ElementType = ElementType> {
+    /**
+    * An HTML element type or a custom React element type to render as.
+    */
+    as?: As;
+}
+
+export interface ChildrenAttribute {
+    children?: ReactNode;
+}
+
+export type AsAttributeWithChildren<As extends ElementType = ElementType> = PropsWithChildren<AsAttribute<As>>;
```

Comment:
I ran small analysis on the code base. Those might not be 100% accurate, but they are close enough i think

86 components total (search on `.displayName = ` in .jsx files)
75 of them have the as property
52 of them as children property
45 have both children and as property
12 have slots

With those stats 

I could make a BaseAttributes interface with only forwarded ref
I could make a BaseWithChildrenAttributes interface with forwarded ref and children
I could make a BaseWithAsAndChildrenAttributes interface with forwarded ref, children and as

Components with as but no children would be Base + AsAttributes, no shared interface to group them
Components with children but no as would be Base + ChildrenAttributes, no shared interface to group them

SlotAttribute would be extended for the 15 components using slots, in addition to one of the 3 base types

Reply:
I need to find appropriate names for those interface if you have any ideas.

Reply:
Also, the Box components would then be 
```
export interface StrictBoxProps extends BaseWithAsAndChildrenAttributes, SlotAttribute
```